File: src/frontend/alunos-client/src/domain/Aluno.ts
export interface Aluno {
  id: number;
  nome: string;
  email: string;
  cpf: string;
  dataNascimento: string | null;
  telefone: string | null;
  endereco: string | null;
  dataCadastro: string;
  ativo: boolean;
}

export interface CreateAlunoDto {
  nome: string;
  email: string;
  cpf: string;
  dataNascimento?: string | null;
  telefone?: string | null;
  endereco?: string | null;
}

export interface UpdateAlunoDto extends CreateAlunoDto {
  ativo: boolean;
}

File: src/backend/AlunosApi/Domain/Entities/Aluno.cs
namespace AlunosApi.Domain.Entities;

public class Aluno
{
    public int Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Cpf { get; set; } = string.Empty;
    public DateTime? DataNascimento { get; set; }
    public string? Telefone { get; set; }
    public string? Endereco { get; set; }
    public DateTime DataCadastro { get; set; } = DateTime.Now;
    public bool Ativo { get; set; } = true;
}